### ISSUE TITLE
feat(datepicker): add DateTimePicker 

### ISSUE THREAD

**Issue by yusijs** [valor-software/ngx-bootstrap#551]
Something similar to https://github.com/Eonasdan/bootstrap-datetimepicker would be immensely useful. I'm not sure if it's an edge case, but I'd love for this to be doable at least. :)

**Comment by jppellerin** [comment]
@yusijs I just implemented something similar.  It's very basic and it makes use of the ng2-bootstrap `datepicker` and `timepicker`.

I don't have this in any commit, but here is the code : 

```
import { Component, OnInit, Input, Self } from '@angular/core';
import { NgModel, ControlValueAccessor } from '@angular/common';
import { TimepickerComponent } from 'ng2-bootstrap';
import { DATEPICKER_DIRECTIVES } from 'ng2-bootstrap/ng2-bootstrap';


@Component({
    selector: 'date-picker',
    template: require('./date-picker.component.html'),
    styles: [require('./date-picker.component.scss')],
    directives: [DATEPICKER_DIRECTIVES, TimepickerComponent]
})

export class DateTimePickerComponent implements ControlValueAccessor, OnInit {

    @Input() milliseconds: boolean;
    public cd: NgModel;
    public onChange: any = Function.prototype;
    public onTouched: any = Function.prototype;
    internalDate: Date;
    internalTime: Date;
    internalDateTime: Date;

    public constructor(@Self() cd: NgModel) {

        this.cd = cd;
        cd.valueAccessor = this;
    }


    ngOnInit() {

        this.internalDate = this.internalDateTime;
        this.internalTime = this.internalDateTime;
    }


    writeValue(obj: any) : void {

        if (obj === this.internalDateTime) {
            return;
        }
        if (obj && obj instanceof Date) {
            this.internalDateTime = obj;
            return;
        }
        this.internalDateTime = obj ? new Date(obj) : void 0;
    }


    registerOnChange(fn: any) : void {

        this.onChange = fn;
    }


    registerOnTouched(fn: any) : void {

        this.onTouched = fn;
    }


    updateDateTime() {

        let newDate: Date = new Date();
        newDate.setTime(this.internalDate.getTime());
        newDate.setHours(this.internalTime.getHours());
        newDate.setMinutes(this.internalTime.getMinutes());
        newDate.setSeconds(this.internalTime.getSeconds());
        newDate.setMilliseconds(this.internalTime.getMilliseconds());
        this.internalDateTime = newDate;
        if (this.milliseconds) {
            this.cd.viewToModelUpdate(newDate.getTime());
        }
        else {
            this.cd.viewToModelUpdate(newDate);
        }
    }


    get date(): Date {

        return this.internalDate;
    }

    set date(date: Date) {

        this.internalDate = date;
        this.updateDateTime();
    }


    get time(): Date {

        return this.internalTime;
    }

    set time(time: Date) {

        this.internalTime = time;
        this.updateDateTime();
    }
}
```

and template looks like this : 

```
<div class="row">
    <div class="col-xs-12">
        <datepicker [(ngModel)]="date" [minDate]="minDate" [showWeeks]="false"></datepicker>
        <timepicker [(ngModel)]="time" [hourStep]="1" [minuteStep]="15" [showMeridian]="false"></timepicker>
    </div>
</div>
```

Usage is done so : 

```
<date-picker *ngIf="isEditable" [(ngModel)]="flyer.startDate" [milliseconds]="true"></date-picker>
```

You could then accept all inputs from both components and pass them along as needed, styling it the way you like, etc.  For the project I am working on this was sufficient and there was no need to do anything more.

Hope this helps!

**Comment by yusijs** [comment]
Thanks - I'll give it a try a bit later today! Do you have the .scss file as well? I'm terrible with styles :3

**Comment by jppellerin** [comment]
@yusijs This is the contents of that file : 

```

```

As you can see... I am also bad with styling.

**Comment by yusijs** [comment]
@jppellerin It's like we're soulmates. :+1:

**Comment by jppellerin** [comment]
I started putting something a bit more official together - hopefully to lead towards a pull request soon-ish.  Pretty busy of late, so I cannot predict when I will be finished.

**Comment by yusijs** [comment]
@jppellerin if you have this on a git repo, I could assist?

**Comment by valorkin** [comment]
Thanks for issue, I was thinkin
Should I include time picker in datepicker or not :)
thanks for an issue! :)

**Comment by yusijs** [comment]
Would be immensely useful. I've made a silly hack now where I bind them to hidden form values, them merge them later in my code. Feels dirty. :p

**Comment by mtfurlan** [comment]
I see that there is a branch for this, what's the status?
Is help needed/useful?

**Comment by touqeershafi** [comment]
any update on this feature ?

**Comment by anil-kk** [comment]
any updates on this feature?

**Comment by Hesesses** [comment]
Any updates?

**Comment by black-byte** [comment]
Any updates on this feature ?

**Comment by fromage9747** [comment]
It's been more than 2 years since this feature request.  Any type of ETA when it will be included?

**Comment by sr9yar** [comment]
I wouldn't mind if there was a dirty hack for this now
But it looks like I'm going to have to quickly write my own dirty hack ...

**Comment by sefasenturk95** [comment]
Still no updates?

**Comment by gongAll** [comment]
Almost 4 years later... Any updates? 
Would be worth noting this is the 5th oldest open issue of this library. ^.^

**Comment by cannontechnology** [comment]
Any updates on this?  I see there are 2 external libraries... Perhaps one can be leveraged?

https://github.com/gillardo/ngx-bootstrap-datetime-popup
https://github.com/Eonasdan/bootstrap-datetimepicker

**Comment by daniloff200** [comment]
Gonna be resolved by #6275, I think
